cmake/templates/HPXMacros.cmake.in

```diff
@@ -11,30 +11,61 @@ include(GNUInstallDirs)
 include(HPX_Utils)
 
 function(hpx_check_compiler_compatibility)
-    if(HPX_IGNORE_COMPILER_COMPATIBILITY OR HPX_WITH_BUILD_BINARY_PACKAGE)
-    return()
-  endif()
 
-  if(NOT (${CMAKE_CXX_COMPILER_ID} STREQUAL ${HPX_CXX_COMPILER_ID}
-     AND ${CMAKE_CXX_COMPILER_VERSION} STREQUAL ${HPX_CXX_COMPILER_VERSION}))
-    set(MESSAGE "Compilers do not match. In order to compile HPX application it is"
-        " recommended to use the same compiler as you did for HPX. "
-        "HPX_CXX_COMPILER=${HPX_CXX_COMPILER}, "
-        "CMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}. "
-        "HPX_CXX_COMPILER_ID=${HPX_CXX_COMPILER_ID}, "
-        "CMAKE_CXX_COMPILER_ID=${CMAKE_CXX_COMPILER_ID}. "
-        "HPX_CXX_COMPILER_VERSION=${HPX_CXX_COMPILER_VERSION}, "
-        "CMAKE_CXX_COMPILER_VERSION=${CMAKE_CXX_COMPILER_VERSION}."
-        "To disable this message set HPX_IGNORE_COMPILER_COMPATIBILITY to On."
-        )
-    if (HPX_FIND_REQUIRED)
-      hpx_error("${MESSAGE}")
+    if(HPX_IGNORE_COMPILER_COMPATIBILITY)
+       return()
+    endif()
+       
+    if(HPX_WITH_BUILD_BINARY_PACKAGE)
+    
+    string(REGEX MATCH "[0-9]+" HPX_MAJOR_VERSION "${HPX_CXX_COMPILER_VERSION}")
+    string(REGEX MATCH "[0-9]+" CXX_MAJOR_VERSION "${CMAKE_CXX_COMPILER_VERSION}")
+   
+        if( NOT (${HPX_MAJOR_VERSION} STREQUAL ${CXX_MAJOR_VERSION}) )
+          set(MESSAGE "Compilers do not match. In order to compile HPX application it is"
+          " recommended to use the same major version of the compiler as you did for HPX. "
+          "HPX_CXX_COMPILER=${HPX_CXX_COMPILER}, "
+          "CMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}. "
+          "HPX_CXX_COMPILER_ID=${HPX_CXX_COMPILER_ID}, "
+          "CMAKE_CXX_COMPILER_ID=${CMAKE_CXX_COMPILER_ID}. "
+          "HPX_CXX_MAJOR_VERSION=${HPX_MAJOR_VERSION}, "
+          "CMAKE_CXX_COMPILER_VERSION=${CXX_MAJOR_VERSION}."
+          "To disable this message set HPX_IGNORE_COMPILER_COMPATIBILITY to On."
+
+        if (HPX_FIND_REQUIRED)
+            hpx_error("${MESSAGE}")
+        else()
+            hpx_warn("${MESSAGE}")
+            set(${CMAKE_FIND_PACKAGE_NAME}_NOT_FOUND_MESSAGE "${MESSAGE}" PARENT_SCOPE)
+            set(${CMAKE_FIND_PACKAGE_NAME}_FOUND False PARENT_SCOPE)
+        endif()
+  
+        endif()
```

Comment:
This should be unindented.

Reply:
As said before, I believe this whole file needs indentation corrections. Something is off.